CondFormats/GeometryObjects/src/PTrackerAdditionalParametersPerDet.cc

```diff
@@ -0,0 +1,50 @@
+#include "CondFormats/GeometryObjects/interface/PTrackerAdditionalParametersPerDet.h"
+#include <iostream>
+
+template <typename T>
+const T getThisParam(const std::vector<std::vector<T>>& params, size_t index_outer, size_t index_inner) {
+  if (index_outer >= params.size())
+    throw std::out_of_range("Parameter with index " + std::to_string(index_outer) + " is out of range.");
+  if (index_inner >= params[index_outer].size())
+    throw std::out_of_range("Parameter with index " + std::to_string(index_inner) + " is out of range.");
+  return params[index_outer][index_inner];
+}
+
+template <typename T>
+void setThisParam(std::vector<std::vector<T>>& params, size_t index_outer, const T& value) {
+  if (index_outer >= params.size())
+    throw std::out_of_range("Parameter with index " + std::to_string(index_outer) + " is out of range.");
+  params[index_outer].push_back(value);
+}
+
+template <typename T>
+const std::vector<T>& getAllParams(const std::vector<std::vector<T>>& params, size_t index_outer) {
+  if (index_outer >= params.size())
+    throw std::out_of_range("Parameter with index " + std::to_string(index_outer) + " is out of range.");
+  return params[index_outer];
+}
```

Comment:
Sorry to come in late, but it would be better to move these to anonymous namespace to avoid adding these somewhat generic (although templated) names into the global namespace.

Reply:
I've wrapped them in `namespace {}` which I think does what you asked for, but if I misunderstood please let me know :-)

Reply:
That was exactly it, thanks!